src/main/java/uk/gov/hmcts/reform/cpo/service/impl/CasePaymentOrdersServiceImpl.java

```diff
@@ -1,9 +1,110 @@
 package uk.gov.hmcts.reform.cpo.service.impl;
 
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
 import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.cpo.data.CasePaymentOrderEntity;
+import uk.gov.hmcts.reform.cpo.exception.CasePaymentOrdersQueryException;
+import uk.gov.hmcts.reform.cpo.repository.CasePaymentOrderQueryFilter;
+import uk.gov.hmcts.reform.cpo.repository.CasePaymentOrdersRepository;
 import uk.gov.hmcts.reform.cpo.service.CasePaymentOrdersService;
+import uk.gov.hmcts.reform.cpo.service.mapper.CasePaymentOrderMapper;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
 
 @Service
 public class CasePaymentOrdersServiceImpl implements CasePaymentOrdersService {
 
+    @Autowired
+    private CasePaymentOrderMapper casePaymentOrderMapper;
+    @Autowired
+    private CasePaymentOrdersRepository casePaymentOrdersRepository;
+
+    public CasePaymentOrdersServiceImpl(CasePaymentOrderMapper casePaymentOrderMapper, CasePaymentOrdersRepository casePaymentOrdersRepository) {
+        this.casePaymentOrderMapper = casePaymentOrderMapper;
+        this.casePaymentOrdersRepository = casePaymentOrdersRepository;
+    }
+
+    public Page<CasePaymentOrderEntity> getCasePaymentOrders(final CasePaymentOrderQueryFilter casePaymentOrderQueryFilter) {
+        if (casePaymentOrderQueryFilter.isItAnEmptyCriteria()) {
+            return Page.empty();
+        }
+        validateCasePaymentOrderQueryFilter(casePaymentOrderQueryFilter);
+
+        final PageRequest pageRequest = getPageRequest(casePaymentOrderQueryFilter);
+        if (casePaymentOrderQueryFilter.isACasesIdQuery()) {
+            return casePaymentOrdersRepository.findByCaseIdIn(
+                casePaymentOrderQueryFilter.getListOfLongCasesIds(),
+                pageRequest
+            );
+        } else {
+            return casePaymentOrdersRepository.findByIdIn(
+                casePaymentOrderQueryFilter.getListUUID(),
+                pageRequest
+            );
+        }
+    }
+
+    private PageRequest getPageRequest(CasePaymentOrderQueryFilter casePaymentOrderQueryFilter) {
+        final List<Sort.Order> orders = new ArrayList<Sort.Order>();
+        orders.add(new Sort.Order(Sort.Direction.ASC,  CasePaymentOrderQueryFilter.CASES_TYPE_ID));
```

Comment:
the scope document says you need to order by case id, not case type id

Reply:
DONE